geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/scala/org/locationtech/geomesa/fs/storage/common/metadata/FileBasedMetadata.scala

```diff
@@ -328,37 +340,65 @@ class FileBasedMetadata(
     }
   }
 
+  /**
+   * Delete a seq of paths
+   *
+   * @param paths paths to delete
+   * @param threads number of threads to use
+   */
+  private def delete(paths: Iterable[Path], threads: Int): Unit = {
+    if (threads < 2) {
+      paths.foreach(fc.delete(_, false))
+    } else {
+      val ec = ExecutionContext.fromExecutorService(new CachedThreadPool(threads))
+      try {
+        val parPaths = paths.par
+        parPaths.tasksupport = new ExecutionContextTaskSupport(ec)
+        parPaths.foreach(fc.delete(_, false))
```

Comment:
Do we need any more error handling here? 

It seems like some deletes may need to be retried.  

Also, is the `paths` variable just the files?  Or does it include directories?  (I.e., can we get a delete called on a directory before it is empty?  And if deleting a file fails the first time, is any attempt to retry it?)

Reply:
paths is just the files in this case, and the recursive flag is passed as false.

i don't think there are transient delete failures that you would be expected to catch and retry... 

generally, `compact` and similar commands are not very robust with regards to failures, probably something we should try to address better, but it's tricky without having anything like transactions.

Reply:
Sounds good!  This as-is is probably ok.

Would it be the case that some operations could sensibly be repeated?  E.g., if a delete hit an issue, could be it re-run and work ok?  (Or maybe I want to that a compact/metadata clean up is repeated?)

Reply:
i can't really think of any transient issues that would likely resolve with a retry, but it's possible... if something does go wrong, then you can use the consistency check commands here to resolve the discrepancies. but yeah, in general we need to be a lot more robust, but that's very hard to pull off

Reply:
Sounds good.  Just asking 8000 questions to see if there are easy ways to make things more robust:).